src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/SelectForUpdateTest.java

```diff
@@ -0,0 +1,177 @@
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.all;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.column;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.entry;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.eq;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.k;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.select;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.table;
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.update;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.ENGINE;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.JDBC;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.PASSWORD;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.SCHEMA_POLICY;
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.USERNAME;
+
+/**
+ * Tests the SELECT ... FOR UPDATE case.
+ *
+ * @author Artur Pedroso (artur.pedroso@feedzai.com)
+ * @since @@@feedzai.next.release@@@
+ */
+@RunWith(Parameterized.class)
+public class SelectForUpdateTest {
+    private static final String TEST_TABLE = "pretty_locks_table";
+    private static final String PK_COL = "pk";
+    private static final String COOL_COL = "change_me";
+    private static final String INIT_COOL_COL_VAL = "init_val";
+    private static final String FINAL_COOL_COL_VAL = "final_val";
+    private static final String PK_COL_VAL = "PK";
+
+
+    protected DatabaseEngine engine;
+    protected Properties properties;
+
+    @Parameterized.Parameters
+    public static Collection<DatabaseConfiguration> data() throws Exception {
+        return DatabaseTestUtil.loadConfigurations();
+    }
+
+    @Parameterized.Parameter
+    public DatabaseConfiguration config;
+
+    @Before
+    public void init() throws Exception {
+        properties = new Properties() {
+
+            {
+                setProperty(JDBC, config.jdbc);
+                setProperty(USERNAME, config.username);
+                setProperty(PASSWORD, config.password);
+                setProperty(ENGINE, config.engine);
+                setProperty(SCHEMA_POLICY, "create");
+            }
+        };
+
+        engine = DatabaseFactory.getConnection(properties);
```

Comment:
I would also add the `this` keyword so its easier to read / understand that this variable is a field when not using an IDE.